components/org.wso2.carbon.identity.media.endpoint/src/main/java/org/wso2/carbon/identity/media/endpoint/service/MediaService.java

```diff
@@ -239,4 +244,55 @@ public void validateMediaType(String mediaTypePathParam, MediaType mediaType) {
         return null;
     }
 
+    /**
+     * Download requested media file.
+     *
+     * @param type       The high level content-type of the resource (if media content-type is image/png then
+     *                   type would be image).
+     * @param id         Unique identifier for the requested media.
+     * @param identifier File identifier.
+     * @return requested media file.
+     */
+    public Response downloadMediaFile(String type, String id, String identifier) {
```

Comment:
shall we add validation for type also? what will happen if type is null?

Reply:
added a validation: https://github.com/wso2/identity-media/pull/19/files#diff-b318058c400489cbe88ca60afe7d6110R308